python/lsst/cp/pipe/ptc.py

```diff
@@ -367,11 +434,269 @@ def runDataRef(self, dataRef, visitPairs):
+
+        dataset.varModel : `dict`
+            Dictionary with modeled variance from data, with amp names as keys.
+
+        dataset.signalAdu : `dict`
+            Dictionary with mean signals in ADU, with amp names as keys.
+
+        dataset.varAdu : `dict`
+            Dictionary with measured variances in ADU^2 from data, with amp names as keys.
+
+        dataset.varModelAdu : `dict`
+            Dictionary with modeled variance in ADU^2 from data, with amp names as keys.
+
+        dataset.a00 : `dict`
+            Dictionary with a00 coefficients from full cov model in Astier+19, with amp names as keys.
+
+        daaset.b00 : `dict`
+            Dictionary with b00 coefficients from full cov model in Astier+19, with amp names as keys.
+        """
+        dataset = OutputPtcDataCovAstierDataset({}, {}, {}, {}, {}, {}, {}, {}, {}, {})
+
+        for i, amp in enumerate(covFits):
+            fit = covFits[amp]
+            meanVecFinal, varVecFinal, varVecModel, wc = fit.getNormalizedFitData(0, 0, divideByMu=False)
+            gain = fit.getGain()
+            dataset.gain.setdefault(amp, []).append(gain)
+            dataset.noise.setdefault(amp, []).append(np.sqrt(fit.getRon()))
+            dataset.signal.setdefault(amp, []).append(meanVecFinal)
+            dataset.var.setdefault(amp, []).append(varVecFinal)
+            dataset.varModel.setdefault(amp, []).append(varVecModel)
+            dataset.signalAdu.setdefault(amp, []).append(meanVecFinal/gain)
+            dataset.varAdu.setdefault(amp, []).append(varVecFinal/(gain**2))
+            dataset.varModelAdu.setdefault(amp, []).append(varVecModel/gain**2)
+            dataset.a00.setdefault(amp, []).append(fit.getA()[0, 0])
+            dataset.b00.setdefault(amp, []).append(fit.getB()[0, 0])
+
+        return dataset
+
+    def computeCovariancesAstier(self, dataRef, visitPairs, detector):
+        """Iterate over pairs of flats to calculate full covariances from the difference image.
+        This follows the treatment in Astier+19
+
+        Parameters
+        ----------
+        dataRef : list of lsst.daf.persistence.ButlerDataRef
+            dataRef for the detector for the visits to be fit.
+
+        visitPairs : `iterable` of `tuple` of `int`
+            Pairs of visit numbers to be processed together
+
+        detector : `lsst.afw.cameraGeom.Detector`
+            Detector object
+
+        Returns
+        -------
+        covariancesWithTags : `numpy.recarray`
+            Covariance measurements.
+        """
+        tupleRecords = []
+        allTags = []
+
+        self.log.info('Measuring Astier covariances using %s visits for detector %s' % (visitPairs,
+                                                                                        detector.getId()))
+
+        for (v1, v2) in visitPairs:
+            # Perform ISR on each exposure
+            dataRef.dataId['expId'] = v1
+            exp1 = self.isr.runDataRef(dataRef).exposure
```

Comment:
Is there a benefit for running ISR here, instead of pulling the postISRCCD data from disk after they've been processed with a particular configuration?  I assume the ISR processing doesn't change, so repeating it from scratch every time a different PTC calculation is done seems like it takes up a lot of the time.

Reply:
I moved the ISR part out of "ptc.py", which now operates on "postISR" images (created by "runIsr.py").